pkg/directory/types/farm.go

```diff
@@ -196,3 +201,12 @@ func FarmUpdate(ctx context.Context, db *mongo.Database, id schema.ID, farm Farm
 	_, err := col.UpdateOne(ctx, f, bson.M{"$set": farm})
 	return err
 }
+
+// FarmIPUpdate update an existing farm ips
+func FarmIPUpdate(ctx context.Context, db *mongo.Database, id schema.ID, ip generated.FarmerIP, workloadID schema.ID) error {
```

Comment:
Does this one even work? this one will override the entire Farm object IP list with this value.

Reply:
I assume you want to update the reservation ID of this ip object to a new value. In that case what u need to do is 
something like that
https://docs.mongodb.com/manual/reference/operator/update/positional/

I really recommend checking out what mongo can do before building more search, or update queries

Reply:
https://docs.mongodb.com/manual/reference/operator/update/positional/

I read the docs and it seems this query should work, because we first query for the IP in the array with a specific value